### PR TITLE
Added support for sending http.client.connect audit events

### PR DESCRIPTION
**Pull request by ravi-kale**
Fixes issue --> https://github.com/urllib3/urllib3/issues/2757

### CODE REVIEW

**Comment by ravi-kale** [comment]
@sethmlarson @pquentin 
Can you please review it, I tried to fix it as much as a unit test I can but still some are still failing due to a segmentation error.

File: test/test_poolmanager.py [hunk 1]
```diff
@@ -474,3 +475,18 @@ def test_thread_safty(self) -> None:
...
         assert conn_queue.qsize() == 0
+
+    @pytest.mark.skipif(not hasattr(sys, "audit"), reason="Requires python 3.8+")
+    def test_http_client_connect_audit_event(self) -> None:
+        event_name = None
+
+        def _hook(event: str, args: tuple) -> None:  # type: ignore[type-arg]
+            if event == "http.client.connect":
+                nonlocal event_name
+                if event == "http.client.connect":
```
Comment (sigmavirus24) [inline test/test_poolmanager.py:486]: Why check this again?

**Review by sigmavirus24 — CHANGES_REQUESTED** [review]

File: test/test_poolmanager.py (hunk 1 above)
Comment (ravi-kale) [reply test/test_poolmanager.py]: Ohh, I missed it, Thanks for review @sigmavirus24

**Review by sigmavirus24 — COMMENTED** [review]
I don't see anything that should be causing segmentation faults. Could you try rebasing against the HEAD of the branch you forked from? That plus a force push would be cleaner than the myriad of merges of main back into this branch

**Comment by ravi-kale** [comment]
> I don't see anything that should be causing segmentation faults. Could you try rebasing against the HEAD of the branch you forked from? That plus a force push would be cleaner than the myriad of merges of main back into this branch

Hey @sigmavirus24, I tried rebasing but still have the same issue, not aware of what caused it to fail. Can you please take a look? That will be helpful.

**Comment by sethmlarson** [comment]
Closed in favor of https://github.com/urllib3/urllib3/pull/2859

**Pull request closed without merging**